_posts/2021-04-27-rust-dev-workflow.md

```diff
@@ -0,0 +1,81 @@
+- The Rust community will use GitHub Issues for tracking feature development and issues, replacing the Jira instance maintained by the Apache Software Foundation (ASF)
+- DataFusion and Ballista will follow a new release cycle, independent of the main Arrow releases
+
+But why, as a community, have we decided to change our processes? Let's take a slightly more in-depth look at the Rust implementation's needs.
+
+## Project Structure
+The Rust implementation of Arrow actually consists of several distinct projects, or in Rust parlance, ["crates"][3]. In addition to the core crates, namely `arrow`, `arrow-flight`, and `parquet`, we also maintain:
+- [DataFusion][4]: an extensible in-memory query execution engine using Arrow as its format
+- [Ballista][5]: a distributed compute platform, powered by Apache Arrow and DataFusion
+
+Whilst these projects are all closely related, with many shared contributors, they're very much at different stages in their respective lifecycles. The core Arrow crate, as an implementation of a spec, has strict compatibility requirements with other versions of Arrow, and this is tested via rigorous cross-language integration tests.
+
+However, at the other end of the spectrum, DataFusion and Ballista are still nascent projects in their own right that undergo frequent backwards-incompatible changes. In the old workflow, DataFusion was released in lockstep with Arrow; because DataFusion users often need newly-contributed features or bugfixes on a tighter schedule than Arrow releases, we observed that many people in the community simply resorted to referencing our GitHub repository directly, rather than properly versioned builds on [crates.io][6], Rust's package registry.
+
+Ultimately, the decision was made to split the Rust crates into two separate repositories: [arrow-rs][7] for the core Arrow functionality, and [arrow-datafusion][8] for DataFusion and Ballista. There's still work to be done on determining the exact release workflows for the latter, but this leaves us in a much better position to meet the broader Rust community's expectations of crate versioning and stability.
+
+## Community Participation
+All Apache projects are built on volunteer contribution; it's a core principle of both the ASF and open-source software development more broadly. One point of friction that was observed in the previous workflow for the Rust community in particular was the requirement for issues to be logged in Arrow's Jira project. This step required would-be contributors to first register an account, and then receive a permissions grant to manage tickets.
+
+To streamline this process for new community members, we've taken the decision to migrate to GitHub Issues for tracking both new development work and known bugs that need addressing, and bootstrapped our new repositories by importing their respective tickets from Jira. We hope that this strikes a better balance between organization and accessibility for prospective contributors.
```

Comment:
It might help to have language that opening issues before PRs are submitted is recommended to let others know you are working on it and discuss technical issues before work is put into a PR.  i.e. this is more a tooling change then anything else.

Reply:
@returnString If you could address this point and update the publish date I think we can go ahead and get this published.

Reply:
Thanks for the feedback @emkornfield! I've incorporated it into that section.

@andygrove I think we're good to go 😄